<launch>
    <!-- use cpp or py file -->
    <arg name="use_py" default="true" />
    <!-- use user argment -->
    <arg name="user" default="false" />

    <!-- follow path config -->
    <arg name="duration" default="0.0" />
    <arg name="distance_tolerance" default="1.0" />

    <!-- csv file path -->
    <arg name="waypoints_name" default="waypoints" />
    <arg name="path_to_waypoints" default="$(find urg_gmapping)/graphs" />

    <!-- run py -->
    <node if="$(arg use_py)" name="run_py" pkg="urg_gmapping" type="run_py.py" output="screen">
        <param name="duration" value="$(arg duration)" />
        <param name="distance_tolerance" value="$(arg distance_tolerance)" />
        <param name="path_to_waypoints" value="$(arg path_to_waypoints)/$(arg waypoints_name).csv" />
    </node>

    <!-- run cpp -->
    <node unless="$(arg use_py)" name="run_cpp" pkg="urg_gmapping" type="run_cpp" output="screen">
        <param name="duration" value="$(arg duration)" />
        <param name="distance_tolerance" value="$(arg distance_tolerance)" />
        <param name="path_to_waypoints" value="$(arg path_to_waypoints)/$(arg waypoints_name).csv" />
    </node>

    <!-- run user -->
    <node if="$(arg user)" name="run_cpp" pkg="urg_gmapping" type="run_cpp" output="screen">
        <param name="duration" value="$(arg duration)" />
        <param name="distance_tolerance" value="$(arg distance_tolerance)" />
        <param name="path_to_waypoints" value="$(arg path_to_waypoints)/user.csv" />
    </node>
</launch>